lib/src/core/client/strategy/strategy.ts

```diff
@@ -205,15 +223,55 @@ export class Strategy {
     return true;
   }
 
-  isEnabled(parameters: any, context: Context): boolean {
+  isEnabled(_parameters: any, _context: Context): boolean {
     return this.returnValue;
   }
 
   isEnabledWithConstraints(
     parameters: any,
     context: Context,
-    constraints: IterableIterator<Constraint | undefined>,
+    constraints: IterableIterator<Constraint | undefined>
   ) {
-    return this.checkConstraints(context, constraints) && this.isEnabled(parameters, context);
+    return (
+      this.checkConstraints(context, constraints) &&
+      this.isEnabled(parameters, context)
+    );
+  }
+
+  getResult(
```

Comment:
this is the same node-client that you have in the dependencies. Any reason why we need to re-implement strategies in next.js?

Reply:
Yes. In Node SDK it's all coupled with `IP` and `UserId` strategy, that can't be implemented in Next.js when using Edge Functions. Since this was a new SDK decision was not to mess with Node. Now that I have more knowledge on the topic and we did client specification, it might be wroth it to try to refactor Node SDK to have dependency injection in strategies.